dev/ctia/dev/split_tests.clj

```diff
@@ -0,0 +1,162 @@
+(ns ctia.dev.split-tests
```

Comment:
Given this is not the first repo that needs "split testing", and that the existing one has been exercised heavily as of lately, I propose we replace this ns with iroh's test runner. Which also fails-fast.

We could extract that code to e.g. https://github.com/threatgrid/clj-experiments (which I intend to be a monorepo for micro projects), ensure it has tests / CI, etc.

I could help out if desired.

Reply:
Sounds good, we can tackle this later.